examples/detail/app.js

```diff
@@ -0,0 +1,153 @@
+    return section;
+  },
+
+  populateItem: function(el, i) {
+    debug('populate item', el);
+    var title = el.firstChild;
+    var body = title.nextSibling;
+    var record = this.getRecordAt(i);
+
+    title.firstChild.data = record.name;
+    body.firstChild.data = record.metadata.artist;
+  },
+
+  populateItemDetail: function(el, i) {
+    var img = el.querySelector('img');
+    img.src = makeImage(i);
+    img.onload = function() {
+      setTimeout(function() {
+        img.classList.add('loaded');
+      });
+    };
+  },
+
+  unpopulateItemDetail: function(el) {
+    var img = el.querySelector('img');
+    img.classList.remove('loaded');
+    img.removeAttribute('src');
+  },
+
+  populateSection: function(el, section, i) {
+    var title = el.firstChild;
+    var height = this.getFullSectionHeight(section);
+    var background = title.nextSibling;
+
+    background.style.height = height + 'px';
+    title.firstChild.data = section;
+  },
+
+  getSections() {
+    debug('get sections');
+    return [];
+  },
+
+  getSectionHeaderHeight() {
+    return 0;
+  },
+
+  getFullSectionHeight() {
+    var result = data.length * itemHeight;
+    debug('full section height', result);
+    return result;
+  },
+
+  getFullSectionLength() {
+    var result = data.length;
+    debug('full section length', result);
+    return result;
+  },
+
+  getSectionFor() {
+    return '';
+  },
+
+  getRecordAt: function(index) {
+    return data[index];
+  },
+
+  getIndexAtPosition: function(pos) {
+    return Math.floor(pos / itemHeight);
+  },
+
+  getPositionForIndex: function(index) {
+    return index * itemHeight;
+  },
+
+  getFullLength: function() {
+    return data.length;
+  },
+
+  getItemHeight: function() {
+    return itemHeight;
+  },
+
+  getFullHeight: function() {
+    return data.length * itemHeight;
+  },
+
+  insertAtIndex: function(index, record, toSection) {
+    return data.splice(index, 0, record);
+  },
+
+  replaceAtIndex: function(index, record) {
+    return data.splice(index, 1)[0];
+  }
+});
+
+
+function createDummyData(count) {
+  var result = [];
+
+  for (var i = 0; i < count; i++) {
+    result.push({
+      name: 'Song ' + i,
+      metadata: {
+        title: 'Song title ' + i,
+        artist: 'Song artist ' + i
+      }
+    });
+  }
+
+  return result;
+}
+
+// Crappy and hopefully expensive canvas code
+function makeImage(index) {
+  var canvas = document.createElement('canvas');
+  canvas.width = canvas.height = 88;
+  var ctx = canvas.getContext('2d');
+  ctx.font = '36px sans-serif';
+  var width = ctx.measureText(index).width
+  var height = ctx.measureText(index).height
+  var top = 26;
+  var left = (88 - width) / 2;
+  ctx.textBaseline = 'top';
+  ctx.fillStyle = '#000'
+  ctx.fillText(index, left - 3, top);
+  ctx.fillStyle = 'red'
+  ctx.fillText(index, left, top);
+  ctx.fillStyle = 'cyan'
+  ctx.fillText(index, left + 3, top);
+  return canvas.toDataURL('image/png');
+}
```

Comment:
Network images will probably require something specific...
Requests could end at any time, and we don't have much control over the expensive steps.
We might need to make `populateItemDetail` return a promise in this case etc...

But I am eager to try out for the music use case in the component.